src/it/scala/io/iohk/ethereum/db/IodbDataSourceIntegrationSuite.scala

```diff
@@ -97,19 +81,16 @@ class IodbDataSourceIntegrationSuite extends FunSuite
       keyList.zip(keyListWithExtraByte).foreach { case (key, value) =>
         assert(dbAfterUpdate.get(OtherNamespace, key).contains(value))
       }
+
+      dbAfterUpdate.close()
     }
   }
 
   test("IodbDataSource insert/update with invalid length") {
     forAll(seqByteStringOfNItemsGen(KeySizeWithoutPrefix)) { unFilteredKeyList: Seq[ByteString] =>
-      //create temporary dir
-      val dir = File.createTempFile("iodbInvalidLength", "iodbInvalidLength")
-      dir.delete()
-      dir.mkdir()
-
       val keyList = unFilteredKeyList.take(KeyNumberLimit)
       val (keysLeft, keysToInsert) = Random.shuffle(keyList).splitAt(Gen.choose(0, keyList.size/2).sample.get)
-      val db = new IodbDataSource(new LSMStore(dir = dir, keySize = KeySize))
+      val db = IodbDataSource(path = "/tmp/iodbInvalidLength", keySize = KeySize)
```

Comment:
Done! Removed `Random.shuffle` from all the tests for which it wasn't necessary, I only left them in a couple of tests for the MPT where it is needed (for example, to see that independently of the order of insertion the hash root of the MPT is the same).